src/PackageCompiler.jl

```diff
@@ -254,16 +262,43 @@ function create_fresh_base_sysimage(stdlibs::Vector{String}; cpu_target::String,
 
         spinner = TerminalSpinners.Spinner(msg = "PackageCompiler: compiling fresh sysimage (incremental=false)")
         TerminalSpinners.@spin spinner begin
-            # Use that to create sys.ji
-            new_sysimage_content = rewrite_sysimg_jl_only_needed_stdlibs(stdlibs)
+            new_sysimage_content = generate_sysimg_jl_contents(stdlibs)    
             new_sysimage_content *= "\nempty!(Base.atexit_hooks)\n"
             new_sysimage_source_path = joinpath(tmp, "sysimage_packagecompiler_$(uuid1()).jl")
             write(new_sysimage_source_path, new_sysimage_content)
+            
+            if VERSION >= v"1.12-"
+                build_settings_source_path = joinpath(tmp, "buildsettings_packagecompiler_$(uuid1()).jl")
+                open(build_settings_source_path, "w") do io
+                    stdlibs_string = join(stdlibs, ',')
+                    println(io, """
+                        INCLUDE_STDLIBS = "$(stdlibs_string)"
+                    """)
+                end
+
+                # The second positional argument `""` is to pass BUILDROOT="" to Base.jl.
+                # If we don't have the "", then Base.jl will assume that one of our other
+                # arguments is the BUILDROOT, which obviously is incorrect.
+                # https://github.com/JuliaLang/PackageCompiler.jl/issues/989
+                #
+                # The --build-settings argument is to pass our BuildSettings file.
+                # BuildSettings is only available in Julia 1.12+
+                # https://github.com/JuliaLang/julia/pull/54387
+                buildsettings_args = `$new_sysimage_source_path "" --build-settings $(build_settings_source_path)`
```

Comment:
I would move all the logic for rewriting the sysimage.jl file into the else branch to show that it is dead on newer julia versions and just use the `Base.find_source_file("sysimg.jl")` as the argument here to show nothing weird is going on.